app/services/slug_setup.rb

```diff
@@ -0,0 +1,58 @@
+class SlugSetup
+    
+    attr_reader :name 
+
+    def self.setup(name)
+        new(name).send(:candidates)
+    end
+
+    def initialize(name)
+        @name = name
+    end
+    
+    def candidates
+        [   @name.method(:short_name), 
+            @name.method(:prolonged_name), 
+            @name.method(:orged), 
+            :name
+        ]
+    end
+    
+end
+
+class String
+    
+    NOT_WANTED = %w(the of for and in to)
+
+    def short_name
+        if self.match(/parochial church/i)
+            parochial_churches
+        else
+            slug_words.first(3).join('-')
+        end
+    end
+
+    def prolonged_name
+        "#{short_name}-#{slug_words[-2]}-#{slug_words[-1]}"
+    end
+
+    def orged
```

Comment:
I find `orged` more fancy, but I agree that `append_org` is more clear :).
I'll change that.

2016-05-09 11:51 GMT+02:00 Sam Joseph <notifications@github.com>:

> In app/services/slug_setup.rb
> <https://github.com/AgileVentures/LocalSupport/pull/292#discussion_r62474304>
> :
>
> > +
> > +    NOT_WANTED = %w(the of for and in to)
> > +
> > +    def short_name
> > +        if self.match(/parochial church/i)
> > +            parochial_churches
> > +        else
> > +            slug_words.first(3).join('-')
> > +        end
> > +    end
> > +
> > +    def prolonged_name
> > +        "#{short_name}-#{slug_words[-2]}-#{slug_words[-1]}"
> > +    end
> > +
> > +    def orged
>
> what do you think to renaming this as suffix_with_org or append_org?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/AgileVentures/LocalSupport/pull/292/files/bf7b03ed1d65dedf261c812acd715258d6344307#r62474304>
>